src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Textbox from '../textbox';
+import I18nHelper from '../../../utils/helpers/i18n';
+import Logger from '../../../utils/logger';
+
+class Decimal extends React.Component {
+  input = null;
+
+  // Create ref to document for tests
+  _document = document;
+
+  formatValue = () => {
+    const { value } = this.props;
+    const { input } = this;
+
+    // Return unformatted value if component has not mounted
+    if (!input) {
+      return value;
+    }
+    // Return unformatted value if input is being used
+    if (input && this._document.activeElement === input.current) {
+      return value;
+    }
+
+    // Only format value if input is not active
+    // Strip delimiters otherwise formatDecimal Helper goes nuts
+    const format = I18nHelper.format();
+    const delimiter = `\\${format.delimiter}`;
+    const delimiterMatcher = new RegExp(`[${delimiter}]*`, 'g');
+    const noDelimiters = value.replace(delimiterMatcher, '');
+
+    return I18nHelper.formatDecimal(
+      noDelimiters,
+      this.validatePrecision()
+    );
+  }
+
+  validatePrecision = () => {
+    const {
+      precision,
+      maxPrecision
+    } = this.props;
+
+    if (precision > maxPrecision) {
+      Logger.warn(`Precision cannot be greater than ${maxPrecision}`);
+      return maxPrecision;
+    }
+
+    return precision;
+  }
+
+  isValidDecimal = (value) => {
+    const { precision } = this.props;
+    const format = I18nHelper.format();
+    const delimiter = `\\${format.delimiter}`;
+    const seperator = `\\${format.separator}`;
+    const validDecimalMatcher = new RegExp(`^[\\d${delimiter}]*[${seperator}{1}]?\\d{0,${precision}}?$`);
+
+    return validDecimalMatcher.test(value);
+  }
+
+  onChange = (evt) => {
+    const { target } = evt;
+    const { value, selectionEnd } = evt.target;
+    const isValid = this.isValidDecimal(value);
+
+    if (isValid) {
+      this.props.onChange(evt);
+    } else {
+      const newPosition = selectionEnd - 1;
+      setTimeout(() => {
+        target.setSelectionRange(newPosition, newPosition);
+      });
+    }
+  }
+
+  onBlur = (evt) => {
+    // Call onChange to force decimal formattting
+    // Alternative is to store input value in state.
+    this.onChange(evt);
```

Comment:
How using `forceUpdate` instead, then it won't trigger `onChange` unnecessarily https://reactjs.org/docs/react-component.html#forceupdate

Reply:
I'd always thought `forceUpdate` was a last resort type thing. Happy to change though. 👍

Reply:
It is, but I think it makes sense in this case.